src/gazebo_mavlink_interface.cpp

```diff
@@ -411,8 +411,8 @@ void GazeboMavlinkInterface::Load(physics::ModelPtr _model, sdf::ElementPtr _sdf
 
   if (_sdf->HasElement("imu_rate")) {
     imu_rate_ = _sdf->GetElement("imu_rate")->Get<int>();
```

Comment:
If you just set` imu_update_interval_ = 1 / _sdf->GetElement("imu_rate")->Get<int>();` in this line you're done - simple and easy to understand. Also because `interval = 1 / rate`, as indicated by the variable names.